File: arbol_Libros.java
import java.util.Scanner;

public class arbol_Libros {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Coleccion de Libros mediante Arbol binario.");
        ArbolBinario_Libro coleccion = new ArbolBinario_Libro();
        int respuesta;
        do {
            Libro libroNuevo = new Libro();
            // ------- ------- ------- Captura
            System.out.println("\nCapture los datos del libro.");
            System.out.println("Titulo del libro: ");
            libroNuevo.setNombre(scan.nextLine());
            System.out.println("Autor: ");
            libroNuevo.setAutor(scan.nextLine());
            System.out.println("Precio de compra: ");
            libroNuevo.setPrecio(scan.nextDouble());
            System.out.println("Fecha de publicacion: ");
            libroNuevo.setPublicacion(scan.nextInt());
            scan.nextLine();
            coleccion.agregar(libroNuevo);
            // ------- control de bucle
            System.out.println("[1] Capturar otro libro. [0] Finalizar.");
            respuesta = scan.nextInt();
            scan.nextLine();
        } while (respuesta != 0);
        // Presentación de datos capturados
        System.out.println("\n[PRE]_______P R E O R D E N_______[PRE]\n");
        coleccion.preorden();
        System.out.println("\n[IN]_______I N O R D E N_______[IN]\n");
        coleccion.inorden();
        System.out.println("\n[POST]_______P O S T O R D E N_______[POST]\n");
        coleccion.postorden();
        // Finalizando
        scan.close();
    }
}

File: arbol_Libros.cpp
#include <iostream>
#include "ArbolBinario_Libro.h"
using namespace std;

Libro capturaLibro(){
    cout << "Captura de datos del libro.\n";
    string tmp_string;
	int tmp_int;
	double tmp_double;
	Libro tmp_libro;
    cout << "Titulo: ";
	getline( cin, tmp_string);
	tmp_libro.setNombre(tmp_string);
    cout << "Autor: ";
	getline( cin, tmp_string);
	tmp_libro.setAutor(tmp_string);
    cout << "Fecha publicacion: ";
	cin >> tmp_int;
	cin.ignore();
	tmp_libro.setPublicacion(tmp_int);
    cout << "Precio compra: ";
	cin >> tmp_double;
	cin.ignore();
	tmp_libro.setPrecio(tmp_double);
    return tmp_libro;
}

int main(){
    arbolBinario misLibros;
    cout << "Arbol binario de Libros.";
    int respuesta;
    do{
        cout << "\n\n----------------------------------------------[CAPTURA]\n";
        Libro nuevoLibro;
        nuevoLibro= capturaLibro();
        // Se agrega al árbol
        misLibros.agregar(nuevoLibro);
        //
        cout << "Oprime 1 continuar, 0 para finalizar= ";
        cin >> respuesta;
        cin.ignore();
    }while(respuesta != 0);
    // Impresiones
    cout << "\n\n---------------------------------------[PRE.ORDEN]\n";
    misLibros.preorden();
    cout << "\n\n---------------------------------------[IN.ORDEN]\n";
    misLibros.inorden();
    cout << "\n\n---------------------------------------[POS.ORDEN]\n";
    misLibros.posorden();
    return 0;
}